--- configure.ac	2008-07-11 04:22:45.000000000 -0400
+++ configure.ac.oden	2008-07-11 04:27:01.000000000 -0400
@@ -487,6 +487,14 @@
     AC_DEFINE(SOCKS5, 1, [Define to enable SOCKS5 firewall-breaching code.])
 fi
 
+
+dnl Check for lib64 target
+dnl FIXME: really check for the target architecture
+case "$host_cpu" in
+x86_64*|sparc64*) lib="lib64";;
+*) lib="lib";;
+esac
+
 AC_MSG_CHECKING(whether to support mysql access.)
 AC_ARG_WITH(mysql,
 [  --with-mysql[=PATH]      Compile with support for mysql access.],
@@ -503,7 +511,7 @@
   *)
     AC_MSG_RESULT(yes)
     if test "x$withval" = "xyes"; then
-      if test -f '/usr/local/lib/mysql/libmysqlclient.a'; then
+      if test -f '/usr/local/$lib/mysql/libmysqlclient.a'; then
 #         AC_ADDLIB("/usr/local/lib/mysql/libmysqlclient.a")
           mysql_dir="/usr/local/bin"
           # Try to find the mysql_config program
@@ -515,7 +523,7 @@
           AC_SUBST(MYSQL_CFLAGS)
           MYSQL_LIBS=$($MYSQL_CONFIG --libs_r)
           AC_SUBST(MYSQL_LIBS)
-      elif test -f '/usr/lib/mysql/libmysqlclient.a'; then
+      elif test -f '/usr/$lib/mysql/libmysqlclient.a'; then
 #        AC_ADDLIB("/usr/lib/mysql/libmysqlclient.a")
          mysql_dir="/usr/bin"
          # Try to find the mysql_config program
@@ -697,10 +705,10 @@
 dnl Checks for libwww and OpenSSL:
 AC_MSG_CHECKING(whether we can find OpenSSL)
 dnl find the ssl library dir (empirical)
-if test -d '/usr/local/ssl/lib'; then
-  ssllib="-L/usr/local/ssl/lib -lssl -lcrypto"
+if test -d '/usr/local/ssl/$lib'; then
+  ssllib="-L/usr/local/ssl/$lib -lssl -lcrypto"
 else
-  ssllib="-L/usr/lib -lssl -lcrypto"
+  ssllib="-L/usr/$lib -lssl -lcrypto"
 fi
 dnl find the ssl include dir (empirical)
 if test -d '/usr/local/ssl/include'; then
